### PR TITLE
Drop CI for OpenSSL 1.0.2

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: src/OpenSSL/SSL.py [hunk 1]
```diff
@@ -28,7 +28,6 @@
     X509Name,
     X509,
     X509Store,
-    X509StoreContext,
```
Comment (reaperhulk) [inline src/OpenSSL/SSL.py:31]: Did we really get rid of this as part of 1.0.2 deprecation? Removing a whole class is a lot... 😮

Comment (alex) [reply src/OpenSSL/SSL.py:31]: It's no longer needed by this module, you can import it from crypto as always.

Comment (reaperhulk) [reply src/OpenSSL/SSL.py:31]: Oh oops, thought this was where it was exported for some reason.

**Comment by reaperhulk** [comment]
As long as our minimum version is cryptography 2.8 (https://github.com/pyca/pyopenssl/blob/master/tox.ini#L13) it seems like we should be supporting 1.0.2 still.

**Comment by alex** [comment]
I'll bump our minimum version them.

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**